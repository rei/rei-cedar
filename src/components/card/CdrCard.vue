<script>
import { defineComponent, useCssModule } from 'vue';

/** Related, interactive containers linking to a single subject or destination */
export default defineComponent({
  name: 'CdrCard',
  props: {
    /**
     * Sets valid HTML container element tag.
     */
    tag: {
      type: String,
      default: 'article',
    },
  },

  setup() {
    return {
      style: useCssModule(),
      baseClass: 'cdr-card',
    };
  },
});
</script>

<template>
  <component
    :is="tag"
    :class="style[baseClass]"
  >
    <!-- @slot CdrCard content -->
    <slot />
  </component>
</template>

<style lang="scss" module src="./styles/CdrCard.module.scss">
</style>
