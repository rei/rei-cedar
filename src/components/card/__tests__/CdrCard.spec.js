import { mount } from '../../../../test/vue-jest-style-workaround.js';
import CdrCard from '../CdrCard.vue';

describe('CdrCard', () => {
  test('renders correctly', () => {
    const wrapper = mount(CdrCard);
    expect(wrapper.element).toMatchSnapshot();
  });
});
