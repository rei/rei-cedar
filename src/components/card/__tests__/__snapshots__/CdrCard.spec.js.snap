// Vitest Snapshot v1

exports[`CdrCard > renders correctly 1`] = `
<article
  class="cdr-card"
>
  <!-- @slot CdrCard content -->
  
  
</article>
`;
